common/src/main/java/org/apache/seata/common/util/ReflectionUtil.java

```diff
@@ -705,6 +705,45 @@ public static Map<String, Object> getAnnotationValues(Annotation annotation) thr
         return getFieldValue(h, "memberValues");
     }
 
+    /**
+     *
+     * @param targetClass the target class
+     * @param methodName the method name
+     * @param paramTypes the param types
+     * @param annotationClass the annotation class
+     * @return the annotation
+     * @param <A>
+     */
+    public static <A extends Annotation> A findAnnotationInHierarchy(
+            Class<?> targetClass, String methodName, Class<?>[] paramTypes, Class<A> annotationClass) {
+        Class<?> superClass = targetClass.getSuperclass();
+        while (superClass != null && superClass != Object.class) {
+            try {
+                Method superMethod = superClass.getDeclaredMethod(methodName, paramTypes);
```

Comment:
Should we attempt to read the annotation from the superclass instead of only checking the method itself?